### PR TITLE
More Windows Store stuff.

### PR DESCRIPTION
**Pull request by sas**

### CODE REVIEW

File: Support/CMake/Toolchain-WinStore.cmake [hunk 1]
```diff
@@ -18,3 +18,8 @@
 # will build for WinStore ARM.
 set(CMAKE_SYSTEM_NAME WindowsStore)
 set(CMAKE_SYSTEM_VERSION 10.0)
+
+# For WinStore we build as a library because creating a process seems
+# restricted, so we just have a separate process open `ds2.dll` and run `main`
+# from it.
+set(LIBRARY 1)
```
Comment (a20012251) [inline Support/CMake/Toolchain-WinStore.cmake]: it looks very clean to have these toolchain files

File: Sources/Utils/Log.cpp [hunk 2]
```diff
@@ -98,9 +98,10 @@ void vLog(int level, char const *classname, char const *funcname,
   functag << funcname;
 
 #if defined(ENABLE_LOGCAT)
-  if (level >= sLogLevel || level >= kLogLevelDebug)
+  if (level >= sLogLevel || level >= kLogLevelDebug) {
     // If we're on Android pollute logcat as well.
     androidLogcat(level, functag.str().c_str(), buffer.data());
+  }
```
Comment (a20012251) [inline Sources/Utils/Log.cpp]: <3

File: Support/CMake/Toolchain-WinStore.cmake (hunk 1 above)
Comment (sas) [inline Support/CMake/Toolchain-WinStore.cmake]: All.

**Comment by a20012251** [comment]
I think we should keep the naming consistent, i.e. having Utils::vsnprintf instead of Utils::VFormat. At least for someone new reading vformat might bring more questions than reading vsnprintf. You might have a better argument though

**Comment by sas** [comment]
I personally prefer having different names (like `ToString` higher in this
file), but more important than my personal preference is the fact that if
we name these functions `snprintf` and `vsnprintf`, the `#pragma poison`
will also warn for uses of these wrappers.

On Tue, Mar 15, 2016 at 2:33 PM, walter erquinigo <notifications@github.com>
wrote:

> I think we should keep the naming consistent, i.e. having Utils::vsnprintf
> instead of Utils::VFormat. At least for someone new reading vformat might
> bring more questions than reading vsnprintf. You might have a better
> argument though
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly or view it on GitHub
> <https://github.com/facebook/ds2/pull/227#issuecomment-197030170>
>



-- 
Stephane Sezer

**Comment by a20012251** [comment]
i was defeated

@facebook-github-bot shipit

**Comment by a20012251** [comment]
you might want to rename the last commit
forgive me please

after that 
@facebook-github-bot pushit

**Comment by a20012251** [comment]
I just can't believe that you had to create your own print function. Life is meaningless

File: Sources/Utils/Log.cpp [hunk 3]
```diff
@@ -151,6 +144,13 @@ void vLog(int level, char const *classname, char const *funcname,
 
   ss << ' ' << buffer.data() << std::endl;
 
+#if defined(OS_WIN32)
+  if (level >= sLogLevel || level >= kLogLevelDebug) {
+    // If we're on Windows, pullute dbgview as well.
```
Comment (a20012251) [inline Sources/Utils/Log.cpp]: pollute

File: Headers/DebugServer2/Utils/String.h [hunk 4]
```diff
@@ -22,6 +27,50 @@ template <typename T> static inline std::string ToString(T val) {
...
+    DS2_ATTRIBUTE_PRINTF(3, 4);
+
+#if defined(OS_WIN32)
+// MSVC does not have snprintf, and has a vsnprintf that does not have the same
+// semantics as the linux one, which means we have to provide wrappers for
+// both.
+static inline int VSNPrintf(char *str, size_t size, char const *format,
+                            va_list ap) {
+  int res = -1;
+  if (size > 0)
```
Comment (a20012251) [inline Headers/DebugServer2/Utils/String.h]: it's your duty

File: Headers/DebugServer2/Utils/String.h [hunk 5]
```diff
@@ -22,6 +27,50 @@ template <typename T> static inline std::string ToString(T val) {
...
+                            va_list ap) {
+  return ::vsnprintf(str, size, format, ap);
+}
+#endif
+
+static inline int SNPrintf(char *str, size_t size, char const *format, ...) {
+  int res;
+  va_list ap;
+  va_start(ap, format);
+  res = VSNPrintf(str, size, format, ap);
```
Comment (a20012251) [inline Headers/DebugServer2/Utils/String.h]: int res = ...

File: Sources/Utils/OptParse.cpp [hunk 6]
```diff
@@ -150,15 +151,33 @@ std::vector<std::string> const &OptParse::getVector(std::string const &name) {
...
-void OptParse::usageDie(std::string const &message) {
-  auto outStream = stderr;
+static void print(char const *format, ...) {
+  va_list ap, sap;
+  std::vector<char> line;
+
+  va_start(ap, format);
+  va_copy(sap, ap);
+
+  size_t size = ds2::Utils::VSNPrintf(nullptr, 0, format, ap) + 1;
```
Comment (a20012251) [inline Sources/Utils/OptParse.cpp]: you know

File: Sources/Utils/OptParse.cpp [hunk 7]
```diff
@@ -150,15 +151,33 @@ std::vector<std::string> const &OptParse::getVector(std::string const &name) {
...
+  line.resize(size);
+  ds2::Utils::VSNPrintf(line.data(), size, format, sap);
 
+  va_end(ap);
+  va_end(sap);
+
+  fputs(line.data(), stderr);
+#if defined(OS_WIN32)
```
Comment (a20012251) [inline Sources/Utils/OptParse.cpp]: curly braces

File: .clang-format [hunk 8]
```diff
@@ -0,0 +1,2 @@
+BasedOnStyle: LLVM
+SortIncludes: true
```
Comment (a20012251) [inline .clang-format]: :(

File: Sources/Utils/OptParse.cpp (hunk 7 above)
Comment (sas) [inline Sources/Utils/OptParse.cpp]: lol

**Comment by a20012251** [comment]
@facebook-github-bot shipit

**Pull request merged**